File: sdk/android/library/src/main/java/com/biglabs/solo/signer/library/models/Wallet.java
package com.biglabs.solo.signer.library.models;

import com.google.gson.annotations.SerializedName;

public class Wallet {
    @SerializedName("id")
    private String id;

    @SerializedName("walletKey")
    private String key;

    @SerializedName("walletId")
    private String address;

    public String getId() {
        return id;
    }

    public String getKey() {
        return key;
    }

    public String getAddress() {
        return address;
    }
}

File: sdk/android/library/src/main/java/com/biglabs/solo/signer/library/models/Wallet.kt
package com.biglabs.solo.signer.library.models

import com.biglabs.solo.signer.library.Coin
import com.google.gson.annotations.SerializedName

class Wallet {
    @SerializedName("id")
    var id: String? = null

    @SerializedName("coin")
    var coinKey: String? = null
    // TODO make coinKey internal

    @SerializedName("network")
    var network: String? = null

    @SerializedName("address")
    var address: String? = null

    fun coin(): Coin? {
        return if (coinKey != null) Coin.fromKey(coinKey!!)!! else null
    }
}
